MediaGalleryMetadata/Model/Jpeg/FileReader.php

```diff
@@ -137,7 +137,10 @@ private function readCompressedImage($resource): string
         } while (!$this->driver->endOfFile($resource));
 
         $endOfImageMarkerPosition = strpos($compressedImage, self::MARKER_PREFIX . self::MARKER_IMAGE_END);
-        $compressedImage = substr($compressedImage, 0, $endOfImageMarkerPosition);
+
+        if ($endOfImageMarkerPosition !== false) {
```

Comment:
in case there is no endMarker do we need to write this marker ?

Reply:
I believe yes, that will actually fix the image format.